modules/mcuboot/hooks/CMakeLists.txt

```diff
@@ -0,0 +1,16 @@
+#
+# Copyright (c) 2022 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+if(CONFIG_BOOT_IMAGE_ACCESS_HOOKS)
+  zephyr_interface_library_named(NRF53HOOKS)
+  zephyr_library()
+  zephyr_library_sources(nrf53_hooks.c)
+  zephyr_library_link_libraries(NRF53HOOKS)
+  target_link_libraries(NRF53HOOKS INTERFACE
+   zephyr_interface
+  )
+  target_link_libraries(NRF53HOOKS INTERFACE MCUBOOT_BOOTUTIL)
```

Comment:
wouldn't it be sufficient to do:
```
   zephyr_library_link_libraries(MCUBOOT_BOOTUTIL)
```
if `nrf53_hooks.c` needs to be compiled with flags from `MCUBOOT_BOOTUTIL` ?

Reply:
applied. @tejlmand What is the difference if I would use `zephyr_link_libraries()` instead of `zephyr_library_link_libraries()`?

Reply:
`zephyr_link_libraries()` will be applied on `zephyr_interface()` and thus ALL zephyr libraries will inherit the link information.

`zephyr_library_link_libraries()` will only be applied, and thus visible, to the current library (and those linking directly with the current library).

So the scope on which `zephyr_library_link_libraries()` is applied is smaller, which is generally a good thing.
Consider it somewhat similar to local / global scoped c variables.